accepted/2022/simplify-rid-model.md

```diff
@@ -0,0 +1,244 @@
+# Simplify RID Model
+
+Date: April 2022
+
+[Runtime IDs (RIDs)](https://docs.microsoft.com/dotnet/core/rid-catalog) are one of the most challenging features of the product. They are necessary, impact many product scenarios, and have very obscure and challenging UX. They also equally affect development and runtime and sometimes prevent .NET from running on operating systems that Microsoft doesn't support. This document proposes a plan to improve some aspects of the way RIDs work, particularly for scenarios that we've found to be insufficient for our needs.
+
+## Context
+
+RIDs are (for the most part) modelled as a graph of [target triplet](https://wiki.osdev.org/Target_Triplet) symbols that describe legal combinations of operating system, chip architecture, and C-runtime, including an extensive fallback scheme. This graph is codified in [`runtime.json`](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.NETCore.Platforms/src/runtime.json), which is describes the RID catalog. It is a massive (database in a) file. That's a lot of data to reason about, hard to edit correctly, and a bear to maintain.
+
+Examples of triplets today are:
+
+- `win-x86`
+- `macos-arm64`
+- `linux-arm64`
+- `linux-x64-musl`
```

Comment:
If this is supposed to be an actual RID (and it feels like it is since the first 3 are RIDs and not triplets), the value is flipped. In .NET RIDs it is: `linux-musl-x64`.

Reply:
Sorry. I always write that wrong. Thanks!

```suggestion
- `linux-musl-x64`
```